src/test/java/io/appform/dropwizard/sharding/dao/RelationalReadOnlyLockedContextTest.java

```diff
@@ -0,0 +1,250 @@
+        sessionFactories.forEach(SessionFactory::close);
+    }
+
+    @Test
+    @SneakyThrows
+    public void testRelationalDaoReadOnlyContext() {
+        Company company1 = Company.builder()
+                .companyUsageId("CMPID1")
+                .companyId(1l)
+                .name("COMP1")
+                .build();
+        Department eng = Department.builder()
+                .name("ENGINEERING")
+                .companyExtId(company1.companyUsageId)
+                .build();
+        Department fin = Department.builder()
+                .name("FINANCE")
+                .companyExtId(company1.companyUsageId)
+                .build();
+        Ceo ceo1 = Ceo.builder()
+                .companyExtId(company1.companyUsageId)
+                .name("KING")
+                .build();
+
+
+        Company company2 = Company.builder()
+                .companyId(2l)
+                .companyUsageId("CMPID2")
+                .name("COMP2")
+                .build();
+        Department hr = Department.builder()
+                .name("HR")
+                .companyExtId(company2.companyUsageId)
+                .build();
+        Ceo ceo2 = Ceo.builder()
+                .companyExtId(company2.companyUsageId)
+                .name("KING-2")
+                .build();
+
+
+        String parentKey = "PARENT_KEY";
+        val lockedContext1 = companyRelationalDao.saveAndGetExecutor(parentKey, company1);
+        lockedContext1.save(departmentRelationalDao, eng1 -> eng);
+        lockedContext1.save(departmentRelationalDao, fin1 -> fin);
+        lockedContext1.save(ceoRelationalDao, ceo -> ceo1);
+        lockedContext1.execute();
+
+        val lockedContext2 = companyRelationalDao.saveAndGetExecutor(parentKey, company2);
+        lockedContext2.save(departmentRelationalDao, hr1 -> hr);
+        lockedContext2.save(ceoRelationalDao, ceo -> ceo2);
+        lockedContext2.execute();
+
+
+        // USE CASE 1 : Using Association Key Annotations
+        val criteria = DetachedCriteria.forClass(Company.class)
+                .add(Restrictions.in("companyId", Sets.newHashSet(company1.getCompanyId(), company2.getCompanyId())));
+        val assosicationSpec = Lists.newArrayList(
+                RelationalDao.QueryAssociationSpec.builder().childMappingKey("companyExtId").parentMappingKey("companyUsageId").build()
+        );
+
+        val dataList = companyRelationalDao.readOnlyExecutor(parentKey, criteria, 0, 4)
+                .readAugmentParent(departmentRelationalDao, null, assosicationSpec, 0, Integer.MAX_VALUE, Company::setDepartments)
+                .readAugmentParent(ceoRelationalDao, null, assosicationSpec, 0, Integer.MAX_VALUE, (parent, childList) -> {
+                    parent.setCeo(childList.stream().findAny().orElse(null));
+                })
+                .execute()
+                .orElse(new ArrayList<>());
+        System.out.println(new ObjectMapper().writeValueAsString(dataList));
```

Comment:
sout does not help in assertions. Add relevant assertions here

Reply:
done